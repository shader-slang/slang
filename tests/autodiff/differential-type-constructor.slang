//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -shaderobj -output-using-type
//TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute -shaderobj -output-using-type

struct MyStruct: IDifferentiable 
{
    float a;
    int b;
}

//TEST_INPUT:ubuffer(data=[0 0 0 0 0], stride=4):out,name=outputBuffer
RWStructuredBuffer<float> outputBuffer;

[numthreads(1, 1, 1)]
void computeMain(uint3 dispatchThreadID: SV_DispatchThreadID)
{
    {
        var dp = diffPair(MyStruct(), MyStruct.Differential());

        outputBuffer[0] = dp.d.a;

        var dp2 = diffPair<MyStruct>(MyStruct(), MyStruct.Differential(1.f));

        outputBuffer[1] = dp2.d.a;
    }
} 