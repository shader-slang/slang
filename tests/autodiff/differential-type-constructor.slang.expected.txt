type: float
0.000000
1.000000
0.000000
0.000000
0.000000
